packages/scripts/config/webpack.config.js

```diff
@@ -102,6 +106,33 @@ const config = {
 					options: getBabelLoaderOptions(),
 				},
 			},
+			{
+				test: /\.(sc|sa|c)ss$/,
```

Comment:
This includes the styles compilation in the javascript ones (relies on `import "style.css"`. That's not how we handle styles in Gutenberg because sometimes a module might want to generate multiple styles. I think we should keep these two separate for now and have something like:

```
wp-scripts build-styles src/style.css src/editor.css src/theme.css -o dist 
```

To create three stylesheets based on the three entry points.

Reply:
I like the idea of having it in a separate script. But don’t you think the usecase where you have multiple style.css files, one for each conponent that get imported in each one to then output one bundles style file would make sense?

Reply:
@youknowriad just wanted to check in and see how you were feeling about my last comment?

Reply:
Hi sorry, I missed the comment.

I'm not sure the proposal defeats the purpose. you can see how we do it in the Gutenberg repo, which is quite similar to what I propose and we already have the use-case you refer to I think.

I understand that it can seem nicer to `import "file.scss"` in a JavaScript file but the downsides here will be the difficulty to control separate outputs aside relying on conventions for filenames or something like that. In a WordPress world, multiple outputs is common. A block for instance can have three stylesheets loaded in three different contexts.

Reply:
But wouldn't that also be the case here? Because we need to specify the entry points for the different files? So therefore we still only rely on the naming convention. I don't fully see how the two are different from that context. 

In the current setup I first test for all the css / sass / scss files, to then compile them based on their original name. So you can literally do whatever you want in regards to the naming convention you want for your project.

And in the proposed setup we would have three scss files that have a fixed name because they are the entry point for Webpack. 

(Sorry if I'm missing something)

Reply:
I just found the error that It only takes the last  imported scss file with a given name. So II might have to switch to the miniExtractCss plugin again and rely on named entry points again. But it would still be the same limitation as the proposed other way of doing init, because you are always relying on the naming convention. 

(Again I might be missing something obvious here  😂)

Thanks for taking the time to go over this with me :)

Reply:
I guess I'm missing something then :) 

Say I have a block that requires three stylesheets:

```js
// block.js

import 'style.scss'; // default styles
import 'editor.scss'; // editor styles
import 'theme.scss'; // theme styles
```

and I'd like to have these three styles in three different stylesheets. How do I do that in this proposal?

Reply:
Just like that.  👍 

With the optimisation splitChunks I am testing the imported scss/sass/css files for their name and channeling them into their respective output file.

The  only promlem I'm still facing it a bug in Webpack [Issue 1967](https://github.com/webpack/webpack/issues/1967). Because of that the script is currently producing unnecessary js versions of each of the css files. 

The only workaround for this that I have found this far, is a cleanup script. But that doesn't seem right. 

Besides these unnecessary files it is looking through all the `scss`/`sass`/`css` imports, running pre and post processors on them and then bundling them into either the `style.css` / `editor.css` / `theme.css` files.  

I would suggest maybe adding a fourth default that is just bundling all the other files in a fourth css file, or throwing an error if a file with another name is imported.

Reply:
(The last comment is referring to the code in the commit (dfa04b9)](https://github.com/WordPress/gutenberg/pull/14847/commits/dfa04b9bd0cfb5208f3c1b06c9fdf5ac24b3af9e)

Reply:
I think this last commit surfaces the issue I was raising. The fact that these "rules" are hard-coded in the webpack config is the issue. What If I'm a theme developper that wants to build two stylesheets for two different pages/templates whatever and I need my own way of splitting the chunks?

For this particular reason I think a CLI command that can take multiple CSS entry points to produce multiple CSS bundles is the way to go.

Reply:
Okay I get that :) But how do we get around that by hardcoding them as entrypoints?

Reply:
I think we shouldn't use webpack to build styles, we should just use the underlying Sass and PostCSS compilers (like we do right now for the Gutenberg packages) https://github.com/WordPress/gutenberg/blob/master/bin/packages/build.js#L118-L155

Reply:
Okay sorry for being so slo to get your point...

I always thought you were talking about using Webpack for it. 

The question is wether I should still be doing it then? I'm okay at writing Webpack configs, but not sure how to best approach this 😮

Reply:
First, I want to thank you for keeping up with this and trying to push it forward. I appreciate it.

My opinion would be to extract the code I just shared as a CLI command in `wp-scripts` with a way to pass args (entrypoints).

Reply:
Cool I'll give it a shot  👍

Reply:
Let me know if you need help or something. @gziolo @nosolosw and others can help here :)

Reply:
I have no idea how this should work but can assist with testing once it's ready :)